crates/rover-client/src/query/schema/push.rs

```diff
@@ -59,20 +59,54 @@ fn build_response(
         return Err(RoverClientError::HandleResponse { msg });
     }
 
-    let hash = match push_response.tag {
-        Some(tag_data) => tag_data.schema.hash,
+    let hash = match &push_response.tag {
+        // we only want to print the first 6 chars of a hash
+        Some(tag_data) => tag_data.schema.hash.clone()[..6].to_string(),
         None => {
             let msg = format!("No schema tag info available ({})", push_response.message);
             return Err(RoverClientError::HandleResponse { msg });
         }
     };
 
+    // If you push the exact same schema as is currently published,
+    // the response CODE is NO_CHANGES but under the result diff,
+    // it gives you the diff for that hash (i.e., the first time it was pushed)
+    // which very well may have changes. For this, we'll just look at the code
+    // first and handle the response as if there was `None` for the diff
+    let change_summary = if push_response.code == "NO_CHANGES" {
+        build_change_summary(None)
+    } else {
+        build_change_summary(push_response.tag.unwrap().diff_to_previous)
```

Comment:
this `unwrap` will panic if `tag` is `None` - we likely want to have an error return here instead

Reply:
@EverlastingBugstopper that should be caught right above in the `match` block, early returning an error if it's `None`